conjur_api/providers/authentication_strategy_factory.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+
+"""
+AuthenticationStrategyFactory module
+This module job is to encapsulate the creation of SSLContext in the dependent of each os environment
+"""
+
+from conjur_api.errors.errors import UnknownAuthnTypeError
+from conjur_api.interface.authentication_strategy_interface import AuthenticationStrategyInterface
+from conjur_api.models.enums.authn_types import AuthnTypes
+from conjur_api.providers.authn_authentication_strategy import AuthnAuthenticationStrategy
+
+# pylint: disable=too-few-public-methods
+def create_authentication_strategy(
```

Comment:
This comment follows from my comments around the authenticate method on the interface.

I think each authenticator has its own set of input parameters and so I think we might need to rework this to take that into account.

Reply:
This is true. The idea would be to add in more named optional parameters to the `create_authentication_strategy` method as we add new authentication strategies.